src/factories/artifactCluster.ts

```diff
@@ -0,0 +1,160 @@
+import { BitmapText, Container } from 'pixi.js'
+import { DEFAULT_ROOT_ARTIFACT_BOTTOM_OFFSET } from '@/consts'
+import { barFactory } from '@/factories/bar'
+import { iconFactory } from '@/factories/icon'
+import { nodeLabelFactory } from '@/factories/label'
+import { waitForApplication, waitForViewport } from '@/objects'
+import { waitForConfig } from '@/objects/config'
+import { emitter } from '@/objects/events'
+import { waitForScale } from '@/objects/scale'
+
+export type ArtifactClusterFactory = Awaited<ReturnType<typeof artifactClusterFactory>>
+
+type ArtifactClusterFactoryRenderProps = {
+  ids: string[],
+  date: Date,
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function artifactClusterFactory() {
+  const application = await waitForApplication()
+  const viewport = await waitForViewport()
+  const config = await waitForConfig()
+  let scale = await waitForScale()
+
+  const element = new Container()
+  const content = new Container()
+  const { element: icon, render: renderIcon } = await iconFactory(true)
+  const { element: label, render: renderLabel } = await nodeLabelFactory(true)
+  const { element: bar, render: renderBar } = await barFactory()
+
+  let currentDate: Date | null = null
+  let currentIds: string[] = []
+
+  element.addChild(bar)
+
+  content.addChild(icon)
+  content.addChild(label)
+  element.addChild(content)
+
+  emitter.on('scaleUpdated', updated => scale = updated)
+
+  application.ticker.add(() => {
+    updatePosition()
```

Comment:
do you know if this is running non stop? And if so can we do something to only run this when the viewport has changed?

Reply:
(these are mostly notes for myself)

I did try and line this up with viewport changes based on the star/end time bindings, but it fell out of sync with guides and got funky, probably because those are debounced or something.

I don't anticipate many of these on the root flow level, but I'll take a second look though and see if something else can be used. I was going to look at assigning the application ticker to the viewport so they're both using the same one, that might help keep things in sync when listening to viewport events as well.

Reply:
I've got something working, will make a separate PR for it.